### DISCUSSION TITLE
How to create a Tailwind PWA prefix?

### DISCUSSION THREAD

**Discussion by Paxatax** [tailwindlabs/tailwindcss#4955] [Help] [answered]
Hello!

I am wanting to create a PWA prefix(is this the right term even?) in Tailwind, where when I add pwa:bg-red-700 (for example), it applies it when viewing the app as a progressive web app.

This would be very similar to Tailwind's dark mode (dark:some-style-700), which gets applied when the dark class is applied to the root element. https://tailwindcss.com/docs/dark-mode

I have the logic for adding pwa class on the root, but am pretty fuzzy on what else would need to be done. I believe I need to add a variant function, then extend all of the styles I want to be affected by this variant.

If anyone has any resources, examples or pointers for getting this done, I'd hugely appreciate it.

Thank you in advance!

**Comment by Paxatax** [comment, marked as answer]
Figured it out. For anyone in need (React):

app.tsx

```
useEffect(() => {
if (
	window.matchMedia &&
	window.matchMedia('(display-mode: standalone)').matches
) {
	document.documentElement.classList.add('pwa');
} else {
	document.documentElement.classList.remove('pwa');
}
}, []);
```


tailwind.config

```
plugin(({ addVariant, e }) => {
addVariant('pwa', ({ modifySelectors, separator }) => {
modifySelectors(({ className }) => {
    return `.pwa .${e(`pwa${separator}${className}`)}`;
    });
});
})
```

**Reply by simonswiss** [reply]
Yep, the correct term here was `variant`, and looks like you found the documentation for adding variants via the plugins API, nice work 👍
